### DISCUSSION TITLE
Request for Comment: How do you use the Node interface?

### DISCUSSION THREAD

**Discussion by cocoahero** [Shopify/storefront-api-feedback#11] [RFC]
Most resource objects in the Storefront API schema currently conform to the `Node` interface. This generally specifies that they have an `id: ID!` field, but also perhaps more importantly opts the resource into being available via the `node(id:)` and `nodes(ids:)` lookup fields on `QueryRoot`.

These fields allows a user to fetch a resource if they already know the correct `ID`. However, because it simply returns an abstract `Node`, users must also know ahead of time what fields to fetch on the polymorphic object using an inline fragment. For example:
```graphql
{
  node(id:"Z2lkOi8vU2hvcGlmeS9Qcm9kdWN0LzE=") {
    ... on Product {
      title
    }
  }
}
```

Our question is simple:  Do you ever use these lookup fields? If so, what use case do they solve for you? Would you prefer something like `product(id: ID)` that removes the need for a polymorphic dance?

**Comment by ahmed-adly-khalil** [comment]
having `product(id: ID)` would be easier to use as most queries that contain node in return objects being filtered out on the client-side, we had a use case that migrates some data once the merchant installs the app and then use webhooks to keep sync. having 2 different schemas was hard to maintain
